src/masonite/orm/collection/Collection.py

```diff
@@ -251,16 +293,42 @@ def _get_value(self, key):
                     items.append(result)
         return items
 
+    def _data_get(self, item, key, default=None):
+        try:
+            if isinstance(item, (list, tuple)):
+                item = item[key]
+            elif isinstance(item, dict):
+                item = item[key]
+            elif isinstance(item, object):
+                item = getattr(item, key)
+        except (IndexError, AttributeError, KeyError, TypeError):
+            return self._value(default)
+
+        return item
+
     def _value(self, value):
         if callable(value):
             return value()
         return value
 
-    def _check_is_callable(self, callback):
+    def _check_is_callable(self, callback, raise_exception=True):
         if not callable(callback):
+            if not raise_exception:
+                return False
             raise ValueError("The 'callback' should be a function")
```

Comment:
change this to should be a callable. It could be a classes method. function is a little wrong here even though a method is technically just a bound function